src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1471,6 +1471,12 @@ void EIRPlantLoopHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
                     condenserNodeType = DataLoopNode::NodeFluidType::Air;
                     condenserNodeConnectionType_Inlet = DataLoopNode::ConnectionType::OutsideAir;
                     condenserNodeConnectionType_Outlet = DataLoopNode::ConnectionType::OutsideAir;
+                    if (sourceSideInletNodeName == sourceSideOutletNodeName) {
+                        ShowSevereError(state, format("PlantLoopHeatPump {} has the same inlet and outlet node.", thisObjectName));
+                        ShowContinueError(state, format("Node Name: {}", sourceSideInletNodeName));
+                        ShowFatalError(state, "Previous condition causes termination.");
+                        errorsFound = true;
+                    }
```

Comment:
IMHO, we should not allow this to happen, given that there is a heat balance at those nodes that would not be available if these nodes used the same node. That said, delete the ShowFatalError here since when errorsFound = true (leave this line here) the program terminates after all EIR HPs are read in after the getInput loop.

    if (errorsFound) {
        // currently there are no straightforward unit tests possible to get here
        // all curves are required and inputs are validated by the input processor
        // obviously this will stay here but I don't feel like counting it against coverage
        ShowFatalError(state, "Previous EIR PLHP errors cause program termination"); // LCOV_EXCL_LINE
    }

Reply:
In addition, we need a way to determine if a dangling node exists (not connected to anything else) so we can correctly report warnings or setup said node as an outdoor air node.

                if (!OutAirNodeManager::CheckOutAirNodeNumber(state, thisFurnace.CondenserNodeNum)) {
                    ShowSevereError(state, format("For {} = {}", CurrentModuleObject, Alphas(1)));
                    ShowContinueError(state, format(" Node name of outdoor dry-bulb temperature sensor not valid outdoor air node= {}", Alphas(14)));
                    ShowContinueError(state, "...does not appear in an OutdoorAir:NodeList or as an OutdoorAir:Node.");
                    ErrorsFound = true;
                }

                OutAirNodeManager::CheckAndAddAirNodeNumber(state, thisChiller.CondInletNodeNum, Okay);

Reply:
If it's tagged as an Inlet node, that already happens in the general node checking.
```
            ShowSevereError(state,
                            format("{}{}{}",
                                   "Node Connection Error, Node=\"",
                                   state.dataBranchNodeConnections->NodeConnections(Loop1).NodeName,
                                   R"(", Inlet node did not find an appropriate matching "outlet" node.)"));
            ShowContinueError(state, "If this is an outdoor air inlet node, it must be listed in an OutdoorAir:Node or OutdoorAir:NodeList object.");
```

Reply:
> In addition, we need a way to determine if a dangling node exists (not connected to anything else) so we can correctly report warnings or setup said node as an outdoor air node.
> 
> ```
>             if (!OutAirNodeManager::CheckOutAirNodeNumber(state, thisFurnace.CondenserNodeNum)) {
>                 ShowSevereError(state, format("For {} = {}", CurrentModuleObject, Alphas(1)));
>                 ShowContinueError(state, format(" Node name of outdoor dry-bulb temperature sensor not valid outdoor air node= {}", Alphas(14)));
>                 ShowContinueError(state, "...does not appear in an OutdoorAir:NodeList or as an OutdoorAir:Node.");
>                 ErrorsFound = true;
>             }
> 
>             OutAirNodeManager::CheckAndAddAirNodeNumber(state, thisChiller.CondInletNodeNum, Okay);
> ```

@rraustad - if we made this change, I think it would need to be considered an input change - even the example file (PlantLoopHeatPump_EIR_AirSource.idf) has the same inlet and outlet node names for the heating and cooling source inlets and outlets

Reply:
in `BranchNodeConnections::CheckNodeConnections`, there is a check for an inlet node that exists elsewhere.  If there is no node match, there is a Severe error (but no fatal):

```
        "   ** Severe  ** Node Connection Error, Node=\"NODE 1\", Inlet node did not find an appropriate matching \"outlet\" node.",
        "   **   ~~~   ** If this is an outdoor air inlet node, it must be listed in an OutdoorAir:Node or OutdoorAir:NodeList object.",
        "   **   ~~~   ** Reference Object=HeatPump:PlantLoop:EIR:Cooling, Name=HP COOLING SIDE",
```

There doesn't seem to be a check to make sure that the outlet node exists elsewhere in the file, but I think this is ok.